airbyte/sources/util.py

```diff
@@ -106,28 +109,49 @@ def get_source(
     metadata: ConnectorMetadata | None = None
     try:
         metadata = get_connector_metadata(name)
-    except exc.AirbyteConnectorNotRegisteredError:
+    except exc.AirbyteConnectorNotRegisteredError as ex:
         if not pip_url:
+            _log_install_state(name, state=EventState.FAILED, exception=ex)
             # We don't have a pip url or registry entry, so we can't install the connector
             raise
 
-    executor = VenvExecutor(
-        name=name,
-        metadata=metadata,
-        target_version=version,
-        pip_url=pip_url,
-    )
-    if install_if_missing:
-        executor.ensure_installation()
+    try:
+        executor = VenvExecutor(
+            name=name,
+            metadata=metadata,
+            target_version=version,
+            pip_url=pip_url,
+        )
+        if install_if_missing:
+            executor.ensure_installation()
 
-    return Source(
-        name=name,
-        config=config,
-        streams=streams,
-        executor=executor,
-    )
+        return Source(
+            name=name,
+            config=config,
+            streams=streams,
+            executor=executor,
+        )
+    except Exception as e:
+        _log_install_state(name, state=EventState.FAILED, exception=e)
+        raise
 
 
 __all__ = [
     "get_source",
 ]
+
+
+def _log_install_state(
+    name: str,
+    state: EventState,
+    exception: Exception | None = None,
+) -> None:
+    """Log an install event."""
+    print(f"{state.value} `{name}` install operation at {pendulum.now().format('HH:mm:ss')}...")
+    send_telemetry(
```

Comment:
I think we might already be printing the install start/stop notification. Need to double-check though...

Reply:
Confirmed, we are printing start/complete messages here:

https://github.com/airbytehq/PyAirbyte/blob/7c70f14b56e68f2d34108da3b5fc2f88cefaec7e/airbyte/_executor.py#L205-L229

What do you think of removing the print from this method?

```suggestion
    send_telemetry(
```